### PR TITLE
add modules to todo

### PR DESCRIPTION
**Pull request by Daniel-Kigozi**
-add functionality to clear completed todos
-added modules to to_do_list

### CODE REVIEW

**Review by Kamba56 — COMMENTED** [review]
Hi @Daniel-Kigozi, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that **does not have complete linters installed**. Kindly install `eslint` also. Please copy the files from [this repo](https://github.com/microverseinc/linters-config/tree/master/html-css-js) anytime you are working on a project that has `HTML`, `CSS` and `js`. Then install webhint, stylelint and eslint.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Daniel-Kigozi** [comment]
Hello, John i was just wondering what you mean when you say my project does
not have linters installed when its the first thing I did from the start of
the project and I even first ran them before I submitted the project.

[image: image.png]

**Comment by Kamba56** [comment]
Hi @Daniel-Kigozi, You have linter installed but they are not complete. You only installed stylelint and webhint which are for CSS and HTML respectively. You need eslint for Javascript since this project uses Javascript. Kindly copy the correct `github/workflow/linters.yml` file and install eslint.

**Review by ha-manel — COMMENTED** [review]
# STATUS: INVALID 🔴 
Hi @Daniel-Kigozi 
I'm sure you put a lot of effort into this project but this PR is invalid because you didn't setup the README file😓 

Kindly fix the issue and submit the PR for review again 👍 
Best of luck!

------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 1]
```diff
@@ -1,166 +1,143 @@
...
+  document.querySelectorAll('.delete').forEach((button) => button.addEventListener('click', (e) => {
+    e.preventDefault();
+    let id;
+    if (button.id > 0) {
+      id = button.id - 1;
+    } else {
+      id = 0;
+    }
+
+    deleteTask(id); // eslint-disable-line
```
Comment (Munsa1) [inline src/index.js]: - [ ] Kindly remove the disable linters line as linters help us to write clean code.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Daniel-Kigozi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

- [ ] The  methods related to the status updates (completed: true / false) should be in a seperate file and should be imported as a module.
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Daniel-Kigozi** [comment]
The linter error occurs bse the two functions call each other so even if i
interchange them they will still bring function called before initialization

**Review by MaganaAsiati — APPROVED** [review]
# Status: Approved [✔️](https://emojipedia.org/check-mark/)
## Hi @Daniel-Kigozi  , I'm @MaganaAsiati , your reviewer for this PR  :wave: 
Great work on making the changes requested by a previous reviewer 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 





## Highlights
- Descriptive pull request :heavy_check_mark:👨🏻‍💻
- Descriptive README file  :heavy_check_mark:👨🏻‍💻
- Great work using the right GitHub Flow.:heavy_check_mark:
- No Linter errors :heavy_check_mark:


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future.




Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
Cheers and Happy coding! 🤓 👏👏👏

**Pull request merged**